modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectAuthenticationHandlerImpl.java

```diff
@@ -145,18 +145,20 @@ public void processAuthenticationResponse(
 
 			JWTClaimsSet jwtClaimsSet = jwt.getJWTClaimsSet();
 
-			JSONObject jsonObject = jwtClaimsSet.toJSONObject();
+			Map<String, Object> claims = jwtClaimsSet.toJSONObject();
 
 			List<String> emails = jwtClaimsSet.getStringListClaim("emails");
 
-			jsonObject.put("email", emails.get(0));
+			claims.put("email", emails.get(0));
 
-			jsonObject.put(
+			claims.put(
 				"family_name", jwtClaimsSet.getStringClaim("family_name"));
-			jsonObject.put(
-				"given_name", jwtClaimsSet.getStringClaim("given_name"));
+			claims.put("given_name", jwtClaimsSet.getStringClaim("given_name"));
 
-			UserInfo userInfo = new UserInfo(jsonObject);
+			UserInfo userInfo = new UserInfo(
+				JSONObjectUtils.parse(
+					com.nimbusds.jose.util.JSONObjectUtils.toJSONString(
+						claims)));
```

Comment:
Hi @marianoalvarosaiz . This seems strange to have to serialize and deserialize. Did you dismiss other approaches?

Reply:
@stian-sigvartsen, the problem is that all return objects with type net.minidev.json.JSONObject have been replaced by java.util.Map<String,Object>, so the only way I find to convert to Map to net.minidev.json.JSONObject is this.

Reply:
See https://github.com/liferay-appsec/liferay-portal/pull/1859